### PR TITLE
feat: exception aggregation

### PR DESCRIPTION
**Pull request by DorisWongNHS**
TIS21-264

### CODE REVIEW

File: src/main/java/uk/nhs/hee/tis/revalidation/integration/router/service/ConnectionServiceRouter.java [hunk 1]
```diff
@@ -52,6 +53,7 @@
   private static final String API_CONNECTION_UNHIDE = "/api/connections/unhide?bridgeEndpoint=true";
   private static final String API_CONNECTION_HIDDEN = "/api/connections/hidden?bridgeEndpoint=true";
   private static final String API_CONNECTION_TCS_HIDDEN = "/api/revalidation/connection/hidden/${header.gmcIds}?searchQuery=${header.searchQuery}&pageNumber=${header.pageNumber}&bridgeEndpoint=true";
+  private static final String API_CONNECTION_TCS_EXCEPION = "/api/revalidation/connection/exception/${header.gmcIds}?searchQuery=${header.searchQuery}&pageNumber=${header.pageNumber}&dbcs=${header.dbcs}&bridgeEndpoint=true";
```
Comment (jayanta2018) [inline src/main/java/uk/nhs/hee/tis/revalidation/integration/router/service/ConnectionServiceRouter.java:56]: Spelling typo "EXCEPION" T is missing

File: src/main/java/uk/nhs/hee/tis/revalidation/integration/router/aggregation/ConnectionExceptionAggregationStrategy.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      result.getMessage().setBody(connectionExceptionDto);
+    } else {
+      final var connectionExceptionDto = aggregateTraineeWithConnectionAllowNull(newExchange);
+      log.info("The exception connections from tcs when reval does not have exception connection :{}", connectionExceptionDto);
+      result.getMessage().setBody(connectionExceptionDto);
+    }
+    return result;
+  }
+
+  private ConnectionHiddenDto aggregateTraineeWithConnection(List<TraineeInfoDto> traineeInfos,
```
Comment (jayanta2018) [inline src/main/java/uk/nhs/hee/tis/revalidation/integration/router/aggregation/ConnectionExceptionAggregationStrategy.java:44]: Is the return type here ConnectionHiddenDto?

File: src/main/java/uk/nhs/hee/tis/revalidation/integration/router/aggregation/ConnectionExceptionAggregationStrategy.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        conn.setConnectionStatus(getConnectionStatus(trainee.getDesignatedBody()));
+      }
+      return conn;
+    }).collect(toList());
+
+    connectionExceptionDto.setConnections(connectionExceptionRecordDtos);
+    return connectionExceptionDto;
+  }
+
+  private ConnectionHiddenDto aggregateTraineeWithConnectionAllowNull(Exchange newExchange) {
```
Comment (jayanta2018) [inline src/main/java/uk/nhs/hee/tis/revalidation/integration/router/aggregation/ConnectionExceptionAggregationStrategy.java:66]: Same here about return type?

File: src/main/java/uk/nhs/hee/tis/revalidation/integration/router/aggregation/ConnectionExceptionAggregationStrategy.java (hunk 2 above)
Comment (DorisWongNHS) [reply src/main/java/uk/nhs/hee/tis/revalidation/integration/router/aggregation/ConnectionExceptionAggregationStrategy.java:44]: It is sharing the same return DTO as Hidden, will refactor the name

File: src/main/java/uk/nhs/hee/tis/revalidation/integration/router/aggregation/ConnectionExceptionAggregationStrategy.java (hunk 3 above)
Comment (DorisWongNHS) [reply src/main/java/uk/nhs/hee/tis/revalidation/integration/router/aggregation/ConnectionExceptionAggregationStrategy.java:66]: it is refactored

File: src/main/java/uk/nhs/hee/tis/revalidation/integration/router/service/ConnectionServiceRouter.java (hunk 1 above)
Comment (DorisWongNHS) [reply src/main/java/uk/nhs/hee/tis/revalidation/integration/router/service/ConnectionServiceRouter.java:56]: Fixed. thank you.

**Pull request closed without merging**